ServerlessLibraryAPI/ClientApp/src/components/DetailView/ActionBar.js

```diff
@@ -1,19 +1,61 @@
 import React, { Component } from "react";
 import { Icon, Link as FabricLink } from "office-ui-fabric-react";
+import { trackEvent } from "../../helpers";
 import "./ActionBar.scss";
 
 class ActionBar extends Component {
+  constructor(props) {
+    super(props);
+    this.outboundRepoClick = this.outboundRepoClick.bind(this);
+    this.outboundDeployClick = this.outboundDeployClick.bind(this);
+    this.openInVSCodeClick = this.openInVSCodeClick.bind(this);
+  }
+
+  outboundDeployClick(template) {
+    let eventData = this.getDeployClickEventDataToLog(template);
+    trackEvent("/sample/deploy/agree", eventData);
+  }
+
+  getDeployClickEventDataToLog(template) {
+    return {
+      template: template
+    };
+  }
+
   getDeployLink(template) {
     return (
       "https://portal.azure.com/#create/Microsoft.Template/uri/" +
       encodeURIComponent(template)
     );
   }
 
+  outboundRepoClick(repository, template) {
+    let eventData = this.getDeployClickEventDataToLog(repository, template);
+    trackEvent("/sample/source", eventData);
+  }
+
+  getDeployClickEventDataToLog(repository, template) {
```

Comment:
need not define separate methods for event data. all the events can have the same data

Reply:
May be in future, different events will need different data